azurerm/internal/services/web/tests/resource_arm_app_service_test.go

```diff
@@ -636,6 +636,171 @@ func TestAccAzureRMAppService_manyIpRestrictions(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMAppService_scmUseMainIPRestriction(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_app_service", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMAppServiceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMAppService_scmUseMainIPRestriction(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAppServiceExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "site_config.0.scm_use_main_ip_restriction", "true"),
```

Comment:
For resources the `data.ImportStep()` checks these values, so this line can be removed safely.